simulation/groups.py

```diff
@@ -0,0 +1,342 @@
+__all__ = [
+    'SGroup',
+    'IGroup',
+    'RGroup',
+    'DGroup'
+]
+
+from simulation import groups_interface
+import random
+
+
+class SGroup(groups_interface.PopulationGroup):
+    def __init__(self, x: float, y: float, label: str, inf_dist: float, infection_prob: float, recover_prob: float,
+                 death_prob: float) -> None:
+
+        super().__init__(x, y, label, inf_dist, infection_prob, recover_prob, death_prob)
+
+    def __del__(self):
+        pass
+
+    def move(self, x_distance: float, y_distance: float, box_width: int,
+             box_height: int) -> bool:
+
+        can_move = True
+
+        if not 0 < self.x + x_distance < box_width:
```

Comment:
ok I can see a reverse logic here with `not` statement, it is ok but in my opinion harder to read